File: rating-service/src/main/java/com/modsen/advice/LoggingAdvice.java
package com.modsen.advice;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Pointcut;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Aspect
@Component
@RequiredArgsConstructor
public class LoggingAdvice {
    private static final Logger log = LoggerFactory.getLogger(LoggingAdvice.class);

    private final ObjectMapper mapper;

    @Around("myPointcut()")
    public Object applicationLogger(ProceedingJoinPoint joinPoint) throws Throwable {
        String methodName = joinPoint.getSignature().getName();
        String className = joinPoint.getTarget().getClass().toString();
        Object[] arrayOfArgument = joinPoint.getArgs();

        log.info("method invoke {}: {}() arguments :{}", className, methodName, mapper.writeValueAsString(arrayOfArgument));
        Object object = joinPoint.proceed();
        log.info("{}: {}() Response", className, methodName);
        return object;
    }

    @Pointcut(value = "execution(* com.modsen.*.*.*(..) ) && !within(com.modsen.config..*)")
    public void myPointcut() {
    }

}

File: promo-code-service/src/main/kotlin/com/modsen/promocodeservice/advice/LoggingAdvice.kt
package com.modsen.promocodeservice.advice

import com.fasterxml.jackson.databind.ObjectMapper
import org.aspectj.lang.ProceedingJoinPoint
import org.aspectj.lang.annotation.Around
import org.aspectj.lang.annotation.Aspect
import org.aspectj.lang.annotation.Pointcut
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Aspect
@Component
class LoggingAdvice(private val mapper: ObjectMapper) {

    private val log: Logger = LoggerFactory.getLogger(LoggingAdvice::class.java)

    @Around("myPointcut()")
    fun applicationLogger(joinPoint: ProceedingJoinPoint): Any {
        val methodName = joinPoint.signature.name
        val className = joinPoint.target.javaClass.toString()
        val arrayOfArgument = joinPoint.args

        log.info("method invoke {}: {}() arguments :{}", className, methodName, mapper.writeValueAsString(arrayOfArgument))
        val result = joinPoint.proceed()
        log.info("{}: {}() Response", className, methodName)
        return result
    }

    @Pointcut("execution(* com.modsen.promocodeservice.*.*.*(..)) && !within(com.modsen.promocodeservice.config..*)")
    fun myPointcut() {
    }
}